libsolidity/ast/Types.cpp

```diff
@@ -508,12 +508,13 @@ BoolResult IntegerType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 	if (_convertTo.category() == category())
 	{
 		IntegerType const& convertTo = dynamic_cast<IntegerType const&>(_convertTo);
-		if (convertTo.m_bits < m_bits)
+		// disallowing unsigned to signed conversion of different bits
+		if (isSigned() != convertTo.isSigned())
+			return false;
```

Comment:
Shouldn't this be allowed if the size is the same?

Reply:
This is for implicit conversion, I think we should not just implicit conversion from say `int8` to `uint8`.

Reply:
Is this still safe to mark as a bugfix?

Reply:
Actually, not sure. This should have been part of `0.8.0`'s strict-conversion rules, so in that sense a bugfix.

Do you think, this is better in 0.9.0?